assets/js/modules/analytics/util/report-validation.js

```diff
@@ -57,3 +57,21 @@ export function isValidDimensions( dimensions ) {
 		return dimension.hasOwnProperty( 'name' ) && typeof dimension.name === 'string';
 	} );
 }
+
+/**
+ * Verifies provided dimensionFilters to make sure they match allowed values found in dimensions.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}   dimensionFilters The dimension filters to check.
+ * @param {Object[]} dimensions       The dimensions to check.
+ * @return {boolean} TRUE if dimension filters are valid, otherwise FALSE.
+ */
+export function isValidDimensionFilters( dimensionFilters, dimensions ) {
```

Comment:
See my other comment, `dimensionFilters` and `dimensions` arguments should not have any relation. `dimensionFilters` are a standalone argument.

Reply:
Updated.